### DISCUSSION TITLE
Tokio Support

### DISCUSSION THREAD

**Discussion by wagenet** [CardinalBytes/daemonize-me#16] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

This doesn't seem to work with tokio. When I try, I get this message:

```
thread 'main' panicked at 'failed to wake I/O driver: Os { code: 9, kind: Uncategorized, message: "Bad file descriptor" }'
```

**Additional context**

This specifically happens when I run `Daemon::start` from a main with `#[tokio::main]`.

**Comment by xadaemon** [comment]
this is likely due to the fork, can you provide platform and os details? Also the used version of the library

**Comment by wagenet** [comment]
tokio 1.16.1, latest version of daemonize-me, macOS 12.3

**Comment by xadaemon** [comment, marked as answer]
Some research tells me you can delay Tokio execution, by making a separate
main function for Tokio. So the way to make demonize-me play along with
Tokio would be make a separate Tokio main function and call it once in the
child process. Because the whole process of demonization is blocking.

**Comment by wagenet** [comment]
Ok, that makes sense. Thanks!

**Comment by xadaemon** [comment]
Do try that, and let me know I will add it to the readme if this works
